AEM/tf-itex-cheat-sheet.md

```diff
@@ -1,244 +1,73 @@
+
+For more information, see [TensorFlow](https://www.tensorflow.org/).
+
+<table><tbody>
+<tr><td>Basic Installation Using PyPI*</td><td>pip install tensorflow</td></tr>
+<tr><td>Basic Installation Using Anaconda*</td><td>conda install -c conda-forge tensorflow</td></tr>
+<tr><td>Import TensorFlow</td><td>import tensorflow as tf</td></tr>
+<tr><td>Capture a Verbose Log (Command Prompt)</td><td>export ONEDNN_VERBOSE=1</td></tr>
+<tr><td>Parallelize Execution (in the Code)</td><td>tf.config.threading.set_intra_op_parallelism_threads(&lt;number of physical cores per socket&gt;) <br> tf.config.threading.set_inter_op_parallelism_threads(&lt;number of sockets&gt;) <br> tf.config.set_soft_device_placement(True)</td></tr>
+<tr><td>Parallelize Execution (Command Prompt)</td><td>export TF_NUM_INTRAOP_THREADS=&lt;number of physical cores per socket&gt; <br> export TF_NUM_INTEROP_THREADS=&lt;number of sockets&gt;</td></tr>
+<tr><td>Non-Uniform Memory Access (NUMA)</td><td>numactl --cpunodebind N --membind N python &lt;script&gt;</td></tr>
+<tr><td>Enable Keras Mixed Precision with BF16</td><td>from tf.keras import mixed_precision <br> mixed_precision.set_global_policy('mixed_bfloat16')</td></tr>
+</tbody></table>
+
+<br><br>
+## Intel® Optimization for TensorFlow*: A Public Release from Intel
+In addition to the performance tuning options listed under the Google public release, the Intel public release offers OpenMP* optimizations for further performance enhancements.
+
+For additional installation methods, see the [Intel® Optimization for TensorFlow* Installation Guide](https://www.intel.com/content/www/us/en/developer/articles/guide/optimization-for-tensorflow-installation-guide.html).
+
+For more information about performance, see the [Maximize TensorFlow* Performance on CPU](https://www.intel.com/content/www/us/en/developer/articles/technical/maximize-tensorflow-performance-on-cpu-considerations-and-recommendations-for-inference.html).
+
+<table><tbody>
+<tr><td>Basic Installation Using PyPI*</td><td>pip install intel-tensorflow</td></tr>
+<tr><td>Basic Installation Using Anaconda*</td><td>conda install tensorflow (Linux/MacOS) <br> conda install tensorflow-mkl (Windows)</td></tr>
+<tr><td>Import TensorFlow</td><td>import tensorflow as tf</td></tr>
+<tr><td>Capture a Verbose Log (Command Prompt)</td><td>export ONEDNN_VERBOSE=1</td></tr>
+<tr><td>Parallelize Execution (in the Code)</td><td>tf.config.threading.set_intra_op_parallelism_threads(&lt;number of physical cores per socket&gt;) <br> tf.config.threading.set_inter_op_parallelism_threads(&lt;number of sockets&gt;) <br> tf.config.set_soft_device_placement(True)</td></tr>
+<tr><td>Parallelize Execution (Command Prompt)</td><td>export TF_NUM_INTRAOP_THREADS=&lt;number of physical cores per socket&gt; <br> export TF_NUM_INTEROP_THREADS=&lt;number of sockets&gt;</td></tr>
```

Comment:
does it work for all TF workloads?  it might be supported only for some TF workloads like TF benchmark.

Reply:
Adding a note to tune the para according to workloads

Reply:
oh. What I mean is "does export TF_NUM_INTEROP_THREADS=X work for all workload?". I guess that it needs to be implement by TF workload itself by " tf.config.set_inter_op_par_threads( os.getenv(TF_NUM_INTEROP_THREADS))"

Reply:
Wonderful. thanks for your finding and explanation.